From 9d5f8dfc6769f4348736e069cd5694e44bf54835 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Thu, 9 Jul 2020 13:37:43 +0800
Subject: [PATCH] policy/modules/system/logging: allow syslogd_t to list user
 temporary directories

Fixes:
avc:  denied  { search } for  pid=109 comm="systemd-journal" name="/"
dev="tmpfs" ino=21007 scontext=system_u:system_r:syslogd_t:s15:c0.c1023
tcontext=root:object_r:user_runtime_t:s0 tclass=dir permissive=1

avc:  denied  { search } for  pid=109 comm="systemd-journal"
name="systemd" dev="tmpfs" ino=21145
scontext=system_u:system_r:syslogd_t:s15:c0.c1023
tcontext=root:object_r:user_tmp_t:s0 tclass=dir permissive=1

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/logging.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/logging.te b/policy/modules/system/logging.te
index 5f29cde41..ae1fb7500 100644
--- a/policy/modules/system/logging.te
+++ b/policy/modules/system/logging.te
@@ -570,6 +570,8 @@ ifdef(`init_systemd',`
 	systemd_manage_journal_files(syslogd_t)
 
 	udev_read_runtime_files(syslogd_t)
+
+	userdom_list_user_tmp(syslogd_t)
 ')
 
 ifdef(`distro_gentoo',`
-- 
2.17.1

